test/unit/gateways/orbital_test.rb

```diff
@@ -108,6 +110,39 @@ def test_successful_purchase
     assert_equal '4A5398CF9B87744GG84A1D30F2F2321C66249416;1', response.authorization
   end
 
+  # Successful Purchase (AC) through Electronic Check as a payment source
+  def test_successful_purchase_with_echeck
+    @gateway.expects(:ssl_post).returns(successful_purchase_with_echeck_response)
+
+    assert response = @gateway.purchase(50, @echeck, order_id: '9baedc697f2cf06457de78')
+    assert_instance_of Response, response
+    assert_equal 'Approved', response.message
+    assert_success response
+    assert_equal '5F8E8BEE7299FD339A38F70CFF6E5D010EF55498;9baedc697f2cf06457de78', response.authorization
+  end
+
+  # Purchase failure due to invalid bank routing number provided in Electronic Check
+  def test_failed_purchase_with_echeck
+    @gateway.expects(:ssl_post).returns(failed_echeck_for_invalid_routing_response)
+
+    assert response = @gateway.purchase(50, @echeck, order_id: '9baedc697f2cf06457de78')
+    assert_instance_of Response, response
+    assert_failure response
+    assert_equal 'Invalid ECP Account Route: []. The field is missing, invalid, or it has exceeded the max length of: [9].', response.message
+    assert_equal '888', response.params['proc_status']
+  end
+
+  # Successful Force Capture (FC) through Electronic Check
+  def test_successful_force_capture_with_echeck
+    @gateway.expects(:ssl_post).returns(successful_force_capture_with_echeck_response)
+
+    assert response = @gateway.purchase(31, @echeck, order_id: '2', force_capture: true)
+    assert_instance_of Response, response
+    assert_equal 'APPROVAL        ', response.message
```

Comment:
Do we need all of this white space behind APPROVAL?

Reply:
Already answered this before.